dss/events/handlers/index.py

```diff
@@ -372,14 +403,37 @@ def notify_subscriber(self, subscription: dict):
                                 f"Code: {response.status_code}")
 
 
-def create_elasticsearch_index(index_name: str, replica: Replica, logger: logging.Logger):
-    es_client = ElasticsearchClient.get(logger)
-    if not es_client.indices.exists(index_name):
-        with open(os.path.join(os.path.dirname(__file__), "mapping.json"), "r") as fh:
-            index_mapping = json.load(fh)
-        index_mapping["mappings"][ESDocType.doc.name] = index_mapping["mappings"].pop("doc")
-        index_mapping["mappings"][ESDocType.query.name] = index_mapping["mappings"].pop("query")
-        alias_name = Config.get_es_alias_name(ESIndexType.docs, replica)
-        create_elasticsearch_doc_index(es_client, index_name, alias_name, logger, index_mapping)
-    else:
-        logger.debug(f"Using existing Elasticsearch index: {index_name}")
+class BundleTombstoneDocument(IndexDocument):
+
+    def __init__(self, replica: Replica, bundle_uuid: str, bundle_version: typing.Optional[str],
+                 logger: logging.Logger) -> None:
+        super().__init__(replica, bundle_uuid, bundle_version, logger)
+
+    @classmethod
+    def from_bucket(cls, replica: Replica, bucket_name: str, key: str, logger):
+        blobstore = Config.get_cloud_specific_handles(replica.name)[0]
+        bundle_uuid, bundle_version = DSS_OBJECT_NAME_REGEX.search(key).groups()
+
+        tombstone_data = json.loads(blobstore.get(bucket_name, key))
+
+        doc = cls(replica, bundle_uuid, bundle_version, logger)
+        doc.update(tombstone_data)
+        return doc
+
+    def list_dead_bundles(self, bucket_name: str):
+        blobstore = Config.get_cloud_specific_handles(self.replica.name)[0]
+
+        if self.bundle_version:
+            # if a version is specified, delete just that version
+            bundle_keys = [bundle_key(self.bundle_uuid, self.bundle_version)]
+        else:
+            # if no version is specified, delete all bundle versions from the index
```

Comment:
We can't assume this is going to finish in the time the lambda runs.  In the interest of not blowing up scope, let's file a separate PR for that and deal with it then.